### PR TITLE
Reimplement cheatsheet using html

### LINKED ISSUE #454: cheat sheet should use imgui.scale setting
I don't even know if it's possible to fetch a setting you haven't defined. But would be really useful for cursorless to use this one

### LINKED ISSUE #505: Displaying help takes a long time
Bringing up the help takes about ten seconds for me.

The following is logged by Talon:
```
2022-01-25 13:49:10 WARNING [watchdog] "user.cursorless-talon.src.cheatsheet.cheat_sheet.draw" @2.0s
   13:                                     talon/canvas.py:17 |
   12:                                     talon/canvas.py:164|
   11:                         talon/scripting/dispatch.py:144|
   10:                         talon/scripting/dispatch.py:135|
    9:                             talon/scripting/rctx.py:233| # 'draw' user.cursorless-talon.src.cheatsheet.cheat_sheet:draw()
    8: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:123| self.draw_section(canvas, "Special mar..
    7: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:161| self.draw_items(canvas, items)
    6: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:236| self.draw_dict(canvas, items)
    5: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:254| self.draw_value(canvas, text)
    4: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:283| canvas.paint.typeface = text_font
    3:                                 talon/skia/paint.py:189|
    2:                                  talon/skia/font.py:79 |
    1:                              talon/skia/typeface.py:44 |
2022-01-25 13:49:13 WARNING [watchdog] "user.cursorless-talon.src.cheatsheet.cheat_sheet.draw" @2.0s
   13:                                     talon/canvas.py:17 |
   12:                                     talon/canvas.py:164|
   11:                         talon/scripting/dispatch.py:144|
   10:                         talon/scripting/dispatch.py:135|
    9:                             talon/scripting/rctx.py:233| # 'draw' user.cursorless-talon.src.cheatsheet.cheat_sheet:draw()
    8: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:123| self.draw_section(canvas, "Special mar..
    7: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:161| self.draw_items(canvas, items)
    6: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:236| self.draw_dict(canvas, items)
    5: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:254| self.draw_value(canvas, text)
    4: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:283| canvas.paint.typeface = text_font
    3:                                 talon/skia/paint.py:189|
    2:                                  talon/skia/font.py:79 |
    1:                              talon/skia/typeface.py:44 |
2022-01-25 13:49:15 WARNING [watchdog] "user.cursorless-talon.src.cheatsheet.cheat_sheet.draw" @2.0s
   13:                                     talon/canvas.py:17 |
   12:                                     talon/canvas.py:164|
   11:                         talon/scripting/dispatch.py:144|
   10:                         talon/scripting/dispatch.py:135|
    9:                             talon/scripting/rctx.py:233| # 'draw' user.cursorless-talon.src.cheatsheet.cheat_sheet:draw()
    8: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:111| self.draw_section(
    7: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:161| self.draw_items(canvas, items)
    6: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:236| self.draw_dict(canvas, items)
    5: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:254| self.draw_value(canvas, text)
    4: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:283| canvas.paint.typeface = text_font
    3:                                 talon/skia/paint.py:189|
    2:                                  talon/skia/font.py:79 |
    1:                              talon/skia/typeface.py:44 |
2022-01-25 13:49:17 WARNING [watchdog] "user.cursorless-talon.src.cheatsheet.cheat_sheet.draw" @2.0s
   13:                                     talon/canvas.py:17 |
   12:                                     talon/canvas.py:164|
   11:                         talon/scripting/dispatch.py:144|
   10:                         talon/scripting/dispatch.py:135|
    9:                             talon/scripting/rctx.py:233| # 'draw' user.cursorless-talon.src.cheatsheet.cheat_sheet:draw()
    8: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:123| self.draw_section(canvas, "Special mar..
    7: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:161| self.draw_items(canvas, items)
    6: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:236| self.draw_dict(canvas, items)
    5: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:254| self.draw_value(canvas, text)
    4: user/cursorless-talon/src/cheatsheet/cheat_sheet.py:283| canvas.paint.typeface = text_font
    3:                                 talon/skia/paint.py:189|
    2:                                  talon/skia/font.py:79 |
    1:                              talon/skia/typeface.py:44 |

```

I am running talon v0.2.0-305-g968a (305)
cursorless-talon 0.25.0
and knausj_talon commit ea9b2d767be59cfccb5e6831874f2173114cd6bb 
on Linux.

Greetings, Anton

### LINKED ISSUE #618: Reimplement cheatsheet using html
We would like to re implement the cheatsheet using html so that we can make it responsive.

### LINKED ISSUE #630: The cheatsheet legend should be moved to more prominent position
The legend is currently located at the bottom left of the screen, where it is easily missed or cut off. 
image

I believe it should be moved to the top left so that it is easily visible.

### PR DESCRIPTION
**Pull request by pokey**
I have left the existing "cursorless help" intact for now, so we can see whether deploy works.  This functionality is all self-contained, and accessible via a new command "cursorless reference"

Fixes #618
Fixes #630
Fixes #454
Fixes #505

### Light mode
image

### Dark mode
image

### Mobile
image

## Todo

- [x] Switch `identifier` to `id`
- [x] Rename `data/cheatsheet`
- [x] Change `spokenForms` to `variations`
- [x] Use camelcase for eg `hat_color` in json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,6 +63,10 @@
 			{
 				"command": "cursorless.resumeRecording",
 				"title": "Cursorless: Resume test case recording"
+			},
+			{
+				"command": "cursorless.showCheatsheet",
+				"title": "Cursorless: Resume test case recording"
```
Comment (Will-Sommers) [inline package.json:69]: see title

File: .gitignore [hunk 2]
```diff
@@ -3,7 +3,7 @@ dist
 node_modules
 .vscode-test/
 *.vsix
-package-lock.json
+/package-lock.json
```
Comment (AndreasArvidsson) [inline .gitignore:6]: Do we have multiple package lock files or why is this changed?

Comment (pokey) [reply .gitignore:6]: Yeah there's one in cursorless-nx subdir now.  We'll merge things when we migrate to nx but for now just wanted to have a lock file in there

File: cursorless-talon/src/cheatsheet_html/cheat_sheet.py [hunk 3]
```diff
@@ -43,39 +46,40 @@ def cursorless_cheat_sheet_get_json():
...
                     "name": "Compound targets",
-                    "id": "compound-targets",
+                    "id": "compoundTargets",
                     "items": get_compound_targets(),
                 },
                 {
                     "name": "Colors",
                     "id": "colors",
-                    "items": get_list("hat_color"),
+                    "items": get_list("hat_color", "hatColor"),
```
Comment (AndreasArvidsson) [inline cursorless-talon/src/cheatsheet_html/cheat_sheet.py:77]: Was this supposed to be a replace?

**Review by AndreasArvidsson — APPROVED** [review]
